Mage/Command/Factory.php

```diff
@@ -43,14 +43,17 @@ public static function get($commandName, Config $config)
             // try a custom command
             $className = 'Command\\' . $commandName;
 
+            // TODO use a custom exception
             if (!class_exists($className)) {
                 throw new Exception('Command "' . $commandName . '" not found.');
             }
         }
 
         /** @var AbstractCommand $instance */
+        // TODO dependencies like $config should be injected into constructor
```

Comment:
Of course, that's what I was talking about. I've wanted only note that not everything in injection mechanisms should be done by constructor. In this part it can be done,even it should be done, because in the other way, `get` method won't work properly. 
And I'm glad too that someone makes such sugestions.